### PR TITLE
Fixes #191390

### LINKED ISSUE #191390: Ranges / Selections Are Difficult To Inspect When Debugging VS Code Extension
Renaming private fields to one character names and properties that are not evaluated automatically make it very difficult to inspect some values, in particular ranges and selections:

Image

I suggest to add more toString implementations, so that the hover becomes more useful.

Also, I wonder why we cannot evaluate getters automatically? I think cdp has an option to run code without side effects.

### PR DESCRIPTION
**Pull request by hediet**
Fixes #191390

Implements `debug.description` for:
* Ranges and Selections: `[10:0 -> 11:12)`
* Position: `(11:12)`
* TextDocument: `TextDocument(file://test.txt)`

These are very common objects and having a nice representation in the debugger makes debugging much more pleasant.

Prior art of formatting ranges:
* [TextSpan.ToString](https://learn.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.text.textspan.tostring?view=roslyn-dotnet-4.7.0): `[10..20)`
* [LinePositionSpan.ToString](https://learn.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.text.linepositionspan.tostring?view=roslyn-dotnet-4.7.0#microsoft-codeanalysis-text-linepositionspan-tostring): `(0,0)-(5,6)`
* VS Code: `[31,1 -> 31,1]` (which refers to columns and thus is a closed interval, wheres the extension host ranges refer to characters and don't include the character at `range.end`)

I think using half-open interval syntax might be more clear for advanced users, while `->` still makes it clear for people who don't know the half-open syntax.

### CODE REVIEW

**Comment by jrieken** [comment]
- [x] check if properties are enum-able
- [ ] more: NotebookDocument, TextEditor, NotebookEditor, ...

File: src/vs/workbench/contrib/chat/test/common/__snapshots__/Response_inline_reference.0.snap [hunk 1]
```diff
@@ -9,15 +9,7 @@
...
-    inlineReference: {
-      scheme: "https",
-      authority: "microsoft.com",
-      path: "/",
-      query: "",
-      fragment: "",
-      _formatted: null,
-      _fsPath: null
-    },
+    inlineReference: URI(https://microsoft.com/),
```
Comment (jrieken) [inline src/vs/workbench/contrib/chat/test/common/__snapshots__/Response_inline_reference.0.snap:12]: Why is this? I though the symbols don't make it into JSON stringified content

File: src/vs/workbench/api/common/extHostTextEditor.ts [hunk 2]
```diff
@@ -566,6 +566,9 @@ export class ExtHostTextEditor {
 			},
 			hide() {
 				_proxy.$tryHideEditor(id);
+			},
+			[Symbol.for('debug.description')]() {
+				return `TextEditor(${this.document.uri.toString()}, ${this.selections.map(s => getDebugDescriptionOfSelection(s)).join(', ')})`;
```
Comment (jrieken) [inline src/vs/workbench/api/common/extHostTextEditor.ts:571]: Unsure, including the selection can be huge and confusing. It makes me also wonder how the equivalent for notebook looks

File: src/vs/workbench/contrib/chat/test/common/__snapshots__/Response_inline_reference.0.snap (hunk 1 above)
Comment (hediet) [reply src/vs/workbench/contrib/chat/test/common/__snapshots__/Response_inline_reference.0.snap:12]: Already created an issue: https://github.com/microsoft/vscode/issues/220048
The debug description symbol is used in `formatValue` to compute the snapshot value...

File: src/vs/workbench/api/common/extHostTextEditor.ts (hunk 2 above)
Comment (hediet) [reply src/vs/workbench/api/common/extHostTextEditor.ts:571]: Lets remove the selection then.

**Review by jrieken — APPROVED** [review]

**Pull request merged**